Source/Libraries/GSF.TimeSeries/Reports/ReportingProcessCollection.cs

```diff
@@ -89,6 +90,56 @@ public virtual string Status
 
         #region [ Methods ]
 
+        /// <summary>
+        /// Loads available <see cref="IReportingProcess"/> implementations.
+        /// </summary>
+        /// <param name="newReportProcessHandler">New report process handler to call when new <see cref="IReportingProcess"/> implementations are loaded.</param>
+        /// <param name="exceptionHandler">Exception handler, if any, to call when type creation fails; otherwise, if <c>null</c> any exceptions will be thrown.</param>
+        public void LoadImplementations(Action<IReportingProcess> newReportProcessHandler, Action<Exception> exceptionHandler = null)
+        {
+            // Manually load known reporting processes as an optimization
+            Add(new CompletenessReportingProcess());
+            Add(new CorrectnessReportingProcess());
+
+            foreach (IReportingProcess reportingProcess in this)
+                newReportProcessHandler(reportingProcess);
+
+            // Load any user defined reporting processes on a background thread
+            new Thread(() =>
```

Comment:
Exceptions thrown from `newReportProcessHandler` or `exceptionHandler` could end up crashing the whole process. Might be good to have a catch-all in the lambda to log exceptions instead.